pkg/controller/kubelet-config/kubelet_config_nodes.go

```diff
@@ -267,6 +273,11 @@ func RunNodeConfigBootstrap(templateDir string, features *osev1.FeatureGate, cco
 	if nodeConfig == nil {
 		return nil, fmt.Errorf("nodes.config.openshift.io resource not found")
 	}
+	// explicitly setting the cgroupMode to "v1"
+	// (TODO) This code can be remove this code in future releases
+	if nodeConfig.Spec.CgroupMode == emptyInput {
+		nodeConfig.Spec.CgroupMode = osev1.CgroupModeV1
```

Comment:
Does this work as we intend? As I understand it, this will only generate the machineconfig with the correct config, but doesn't actually create the nodeConfig object to cluster.

This combined with the above `ctrl.configClient.ConfigV1().Nodes().Update(context.TODO(), nodeConfig, metav1.UpdateOptions{})`, is it possible that we would run into some race condition where the MCO generates the machineconfigs with an empty nodes.config before the above sync has a chance to run?

Or, hmm, maybe I am overthinking it. Just wanted to make sure

Reply:
> Does this work as we intend? As I understand it, this will only generate the machineconfig with the correct config, but doesn't actually create the nodeConfig object to cluster.

The nodeConfig object is by default created, present in the cluster.

> This combined with the above ctrl.configClient.ConfigV1().Nodes().Update(context.TODO(), nodeConfig, metav1.UpdateOptions{}), is it possible that we would run into some race condition where the MCO generates the machineconfigs with an empty nodes.config before the above sync has a chance to run?

Yeah, our goal here is to mainly update the config node object with the `cgroupv1` context. Creation of the respective machineconfig will only add the `kernelArgs` to enable `cgroupsv1` which is already taken care as part of the base templates (explicit setting of cgroupsv1). So, we need not worr about running into any race condition.

Reply:
> The nodeConfig object is by default created, present in the cluster.

Right, the default used to be empty.

> Yeah, our goal here is to mainly update the config node object with the cgroupv1 context. Creation of the respective machineconfig will only add the kernelArgs to enable cgroupsv1 which is already taken care as part of the base templates (explicit setting of cgroupsv1). So, we need not worr about running into any race condition.

Ah, so you verified that the base kernalArgs and the kernalArgs generated by setting the nodes.config object is the **exact same** ?

I was concerned about our karg merging logic since I don't remember if it was correct or not. If you can verify that it is exact same and no extra rendered config needs to be generated, then I think we are good to go

Reply:
Yes, the kArgs are the same but there would be a new rendered MC as there is a `97-{master/worker}-generated-kubelet` are created.